pkg/cluster/delete.go

```diff
@@ -30,17 +30,33 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
-func (m *manager) deleteNic(ctx context.Context, resource mgmtfeatures.GenericResourceExpanded) error {
+// deleteNic deletes the network interface resource by first fetching the resource using the interface
+// client, checking the provisioning state to ensure it is 'succeeded', and then deletes it
+// If the nic is in a failed provisioning state, it will perform an empty CreateOrUpdate on it to put it back into
+// a succeeded provisioning state.
+//
+// The resources client incorrectly reports provisioningState hence we must use the interface client to fetch
+// this resource again so we get the correct provisioningState instead of always just "Succeeded"
+func (m *manager) deleteNic(ctx context.Context, nicName string) error {
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
 
-	if resource.ProvisioningState != nil && !strings.EqualFold(*resource.ProvisioningState, "succeeded") {
-		m.log.Printf("NIC '%s' is not in a succeeded provisioning state, attempting to reconcile prior to deletion.", *resource.ID)
-		err := m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, *resource.Name, mgmtnetwork.Interface{})
+	nic, err := m.interfaces.Get(ctx, resourceGroup, nicName, "")
```

Comment:
nit: should not we rename interfaces to networkInterfaces or something like that ? like, the "interfaces" make it sound like it is some generic interface for all types of resources from any rp.

Reply:
In `azureactions` we use `networkInterfaces`.  Will add in a follow-up PR.